### DISCUSSION TITLE
How to use ConsistentSampler with auto instrumentation

### DISCUSSION THREAD

**Discussion by kes2464** [open-telemetry/opentelemetry-java-contrib#999] [Q&A] [answered]
Hi @oertl,

I am new to contrib repo and autoconfigure SPI (also quite new to OpenTelemetry itself), sorry in advance if I am asking an obvious question.

I am trying to use ConsistentSampler with auto instrumentation.
It seems like ConsistentSamplers are missing ConfigurableSamplerProviders in contrib repo.

I could easily add a provider in contrib repo to use it in auto instrumentation, but what is the expectation here? Should I create another project just for myself to use which has a dependency on opentelemetry-consistent-sampling and has my own ConfigurableSamplerProviders? Or Can I or someone actually add it in contrib repo?

Thanks

**Comment by trask** [comment, marked as answer]
> Can I or someone actually add it in contrib repo?

hi @kes2464! this sounds reasonable to me, go ahead and send an initial PR and we can discuss details there

**Reply by kes2464** [reply]
Hi @trask , I have created a PR https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1005. I have only added one provider which I need, but I am happy to discuss further if we should actually add them all. Thanks
